Code/Framework/AzCore/AzCore/Serialization/SerializeContext.h

```diff
@@ -2572,7 +2567,7 @@ namespace AZ
             return static_cast<GenericClassInfoType*>(findIt->second);
         }
 
-        void* rawMemory = m_moduleOSAllocator.Allocate(sizeof(GenericClassInfoType), alignof(GenericClassInfoType));
+        void* rawMemory = AllocatorInstance<OSAllocator>::Get().Allocate(sizeof(GenericClassInfoType), alignof(GenericClassInfoType));
```

Comment:
It can use either one. This is inserted into `m_moduleLocalGenericClassInfos`, which is a `AZStd::unordered_map<AZ::Uuid, AZ::GenericClassInfo*>`. That type's allocator ends up routing to the `SystemAllocator`, but that describes how the container itself allocates its internal data, not how the mapped type should be allocated. It's the same as having a `vector<SomeType>` where `SomeType` is defined to use the `PoolAllocator`. Since this was already being allocated with the `OSAllocator`, I kept that part.

Reply:
I've gone ahead and updated this to use `azcreate`, which will allocate the object using the `SystemAllocator`.